zephyr/zephyr_reporter.py

```diff
@@ -136,46 +136,47 @@ def update_zephyr(test_cases_list):
-
-            if test_id:
-                print "Found Test ID in Meta file : " + test_id
-                issue = jira.issue(test_id)
-            else:
-                continue
-
-            if z["updateautomationstatus"]:
-                 update_automation_status(issue)
-
-            exec_id = reporter.create_execution(str(issue.id))
-            result = test_cases_list[i][1]
-            print "Test case Result: " + result
-            log_data = "sample log data"
-            if result == 'FAIL':
-                result = 'FAIL'
-            if result == 'OK':
-                result = 'PASS'
-            if result == 'None':
-                result = 'FAIL'
-            if result == 'SKIP':
-                result = 'NOT TESTED'
-            if result == 'Exp FAIL':
-                result = 'FAIL'
-
-            if 'status_codes' in z:
-                ret = reporter.set_execution(result,
-                 exec_id,
-                 log_data,
-                 status_code_dict=z['status_codes'])
-            else:
-                ret = reporter.set_execution(result,
-                 exec_id,
-                 log_data)
-
-            if ret.status_code != requests.codes.ok:
-                raise Exception("Error = %s, when trying to set execution status" % ret)
+            if test_cases_list[i] != []:
+                test_name = test_cases_list[i][0]
+                print "Test_name :" + test_name
+                test_id = get_test_id_from_meta_file(z["metafile"], test_name)
+
+                if test_id:
+                    print "Found Test ID in Meta file : " + test_id
+                    issue = jira.issue(test_id)
+                else:
+                    continue
+
+                if z["updateautomationstatus"]:
+                     update_automation_status(issue)
+
+                exec_id = reporter.create_execution(str(issue.id))
+                result = test_cases_list[i][1]
+                print "Test case Result: " + result
+                log_data = "sample log data"
+                if result == 'FAIL':
+                    result = 'FAIL'
+                if result == 'OK':
+                    result = 'PASS'
+                if result == 'None':
+                    result = 'FAIL'
+                if result == 'SKIP':
+                    result = 'NOT TESTED'
+                if result == 'Exp FAIL':
+                    result = 'FAIL'
+
+                if 'status_codes' in z:
+                    ret = reporter.set_execution(result,
+                     exec_id,
+                     log_data,
+                     status_code_dict=z['status_codes'])
+                else:
+                    ret = reporter.set_execution(result,
+                     exec_id,
+                     log_data)
+
+                if ret.status_code != requests.codes.ok:
+                    raise Exception("Error = %s, when trying to set execution status" % ret)
```

Comment:
Is this really needed when the other change merged? 

https://github.com/lgirdk/boardfarm/commit/36b8fac7872ea11d9b2495451f360c83b1327cfd

Reply:
I don't know what other change merged? Can you tell more detail for this?
This commit is to fix if test_cases_list[i] ==[] will error on line 139

Reply:
I think you just need to fix the for loop:

for test_case in test_cases_list:

should be 

for i in range(len(test_cases_list)):